sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java

```diff
@@ -185,32 +204,311 @@ HttpPipeline getHttpPipeline() {
+     *
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName}
+     * are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<KeyVaultRoleDefinition> setRoleDefinition(KeyVaultRoleScope roleScope) {
+        return setRoleDefinition(new SetKeyVaultRoleDefinitionOptions(roleScope, UUID.randomUUID().toString()));
+    }
+
+    /**
+     * Creates or updates a {@link KeyVaultRoleDefinition}.
+     *
+     * @param roleScope The {@link KeyVaultRoleScope role scope} of the {@link KeyVaultRoleDefinition}. Managed HSM only
+     * supports '/'.
+     * @param roleDefinitionName The name of the {@link KeyVaultRoleDefinition}. It can be any valid UUID.
+     *
+     * @return A {@link Mono} containing the created {@link KeyVaultRoleDefinition}.
+     *
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName}
+     * are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<KeyVaultRoleDefinition> setRoleDefinition(KeyVaultRoleScope roleScope, String roleDefinitionName) {
+        return setRoleDefinition(new SetKeyVaultRoleDefinitionOptions(roleScope, roleDefinitionName));
+    }
+
+    /**
+     * Creates or updates a {@link KeyVaultRoleDefinition}.
+     *
+     * @param options Object representing the configurable options to create or update a
+     * {@link KeyVaultRoleDefinition role definition}.
+     *
+     * @return A {@link Mono} containing the created or updated {@link KeyVaultRoleDefinition}.
+     *
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName}
+     * in the {@link SetKeyVaultRoleDefinitionOptions options} object are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<KeyVaultRoleDefinition> setRoleDefinition(SetKeyVaultRoleDefinitionOptions options) {
+        return setRoleDefinitionWithResponse(options).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Creates or updates a {@link KeyVaultRoleDefinition}.
+     *
+     * @param options Object representing the configurable options to create or update a
+     * {@link KeyVaultRoleDefinition role definition}.
+     *
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the
+     * created or updated {@link KeyVaultRoleDefinition}.
+     *
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName}
+     * in the {@link SetKeyVaultRoleDefinitionOptions options} object are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<KeyVaultRoleDefinition>> setRoleDefinitionWithResponse(
+        SetKeyVaultRoleDefinitionOptions options) {
+
+        return withContext(context -> setRoleDefinitionWithResponse(options, context));
+    }
```

Comment:
We can remove `setRoleDefinition(SetKeyVaultRoleDefinitionOptions)` in favor of the `WithResponse` variant.